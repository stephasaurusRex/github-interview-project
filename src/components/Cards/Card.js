import React, { Component } from "react";

class Card extends Component {
  render() {
    const { repo, rank } = this.props;
    const side = rank <= 50 ? 'first-half' : 'second-half';
    return (
      <div className={"card grid-item " + side}>
        <div className='repo-rank'>{rank}</div>
        <div className="repo-title">
          <a target="_blank" href={repo ? repo.svn_url : 'Loading Repo...'}>{repo ? repo.name : ''}</a>
        </div>
        <div className="repo-stars">
          <i className="fa fa-star" aria-hidden="true"></i>
          {repo ? repo.stargazers_count : ''}
          <i className="fa fa-star" aria-hidden="true"></i>
          </div>
        <div className="top-contributor">
          <img className='avatar-img' src={
            repo &&
            repo.topContributor &&
            repo.topContributor.avatar_url ?
              repo.topContributor.avatar_url :
              'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSCuBm8iZkWaoTJ5ruwULeLtYXaExdy3SpmhqkaDkADI0XJP26oZA'}
               alt={ repo && repo.topContributor ? repo.topContributor.login : ''}
               width="30"
               height="30"/>
          <a target="_blank" className="yellow-link" href={repo && repo.topContributor ? repo.topContributor.html_url : ''}>
            {repo && repo.topContributor ? repo.topContributor.login : 'Loading Contributor....'}
            </a>
        </div>
      </div>
    )

  }
}

export default Card;
