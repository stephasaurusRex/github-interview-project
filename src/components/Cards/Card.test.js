import "jest-styled-components";

import React from "react";
import TestRenderer from "react-test-renderer";

import Card from "./Card";

describe("Card", () => {

  it("renders without props", () => {
    const testRenderer = TestRenderer.create(
      <Card/>
    );
    expect(testRenderer).toMatchSnapshot();
    testRenderer.unmount();
  });

  it("renders with props", () => {
    const repo = {
      name: 'Test Repo',
      svn_url: 'https://testthatcard.com',
      stargazers_count: 14,
      topContributor: {
        login: 'barry manilow',
        html_url: 'http://thisisnotreal.com',
        avatar_url: 'https://google.com',
      }
    }

    const testRenderer = TestRenderer.create(
      <Card key="123" rank="456" repo={repo}/>
    );
    expect(testRenderer).toMatchSnapshot();
  });

  it("renders error with insufficient props", () => {
    const repo = {
      name: 'Test Repo',
      svn_url: 'https://testthatcard.com',
      stargazers_count: 14,
    }

    const testRenderer = TestRenderer.create(
      <Card key="123" rank="456" repo={repo}/>
    );
    expect(testRenderer).toMatchSnapshot();
  });
});
