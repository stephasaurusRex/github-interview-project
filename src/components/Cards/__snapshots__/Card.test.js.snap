// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders error with insufficient props 1`] = `
<div
  className="card grid-item second-half"
>
  <div
    className="repo-rank"
  >
    456
  </div>
  <div
    className="repo-title"
  >
    <a
      href="https://testthatcard.com"
      target="_blank"
    >
      Test Repo
    </a>
  </div>
  <div
    className="repo-stars"
  >
    <i
      aria-hidden="true"
      className="fa fa-star"
    />
    14
    <i
      aria-hidden="true"
      className="fa fa-star"
    />
  </div>
  <div
    className="top-contributor"
  >
    <img
      alt=""
      className="avatar-img"
      height="30"
      src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSCuBm8iZkWaoTJ5ruwULeLtYXaExdy3SpmhqkaDkADI0XJP26oZA"
      width="30"
    />
    <a
      className="yellow-link"
      href=""
      target="_blank"
    >
      Loading Contributor....
    </a>
  </div>
</div>
`;

exports[`Card renders with props 1`] = `
<div
  className="card grid-item second-half"
>
  <div
    className="repo-rank"
  >
    456
  </div>
  <div
    className="repo-title"
  >
    <a
      href="https://testthatcard.com"
      target="_blank"
    >
      Test Repo
    </a>
  </div>
  <div
    className="repo-stars"
  >
    <i
      aria-hidden="true"
      className="fa fa-star"
    />
    14
    <i
      aria-hidden="true"
      className="fa fa-star"
    />
  </div>
  <div
    className="top-contributor"
  >
    <img
      alt="barry manilow"
      className="avatar-img"
      height="30"
      src="https://google.com"
      width="30"
    />
    <a
      className="yellow-link"
      href="http://thisisnotreal.com"
      target="_blank"
    >
      barry manilow
    </a>
  </div>
</div>
`;

exports[`Card renders without props 1`] = `
<div
  className="card grid-item second-half"
>
  <div
    className="repo-rank"
  />
  <div
    className="repo-title"
  >
    <a
      href="Loading Repo..."
      target="_blank"
    >
      
    </a>
  </div>
  <div
    className="repo-stars"
  >
    <i
      aria-hidden="true"
      className="fa fa-star"
    />
    
    <i
      aria-hidden="true"
      className="fa fa-star"
    />
  </div>
  <div
    className="top-contributor"
  >
    <img
      alt=""
      className="avatar-img"
      height="30"
      src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSCuBm8iZkWaoTJ5ruwULeLtYXaExdy3SpmhqkaDkADI0XJP26oZA"
      width="30"
    />
    <a
      className="yellow-link"
      href=""
      target="_blank"
    >
      Loading Contributor....
    </a>
  </div>
</div>
`;
